modules/tm/async.c

```diff
@@ -249,7 +249,7 @@ int t_handle_async(struct sip_msg *msg, struct action* a,
 	} else {
 		/* update the cloned UAS (from transaction)
 		 * with data from current msg */
-		if (t->uas.request)
+		if ((t->uas.request) && (route_type==REQUEST_ROUTE) && ((msg->msg_flags & FL_TM_FAKE_REQ) == 0))
```

Comment:
Should this be just `if ((t->uas.request) && !(msg->msg_flags & FL_TM_FAKE_REQ))` ?

Reply:
While it _seems_ we could enable more scenarios by omitting the `(route_type==REQUEST_ROUTE)` filtering as you suggest (after all, we only need `!is_faked_req()` here) , the script's async calls are not designed (nor much tested) beyond the `request_route`.  This design choice can even be seen in the same function, a [few lines below](https://github.com/OpenSIPS/opensips/blob/master/modules/tm/async.c#L321), where any route apart from `REQUEST_ROUTE` is simply handled as sync, to be on the safe side.  The restriction is also documented on the opensips.org wiki, [here](https://www.opensips.org/Documentation/Script-Async-3-6#async_routes).

So, following the same principle, I suggest we keep the filter, to avoid potential issues with non-faked requests from non-request routes.  Maybe an `async()` in `local_route` or `event_route` falls in this category?!